environment.py

```diff
@@ -99,9 +99,13 @@ def _create_signalfx_ingest(self):
         :return: Ingest Client
 
         """
-        client = signalfx.SignalFx()
-        ingest = client.ingest(self._ingest_token, endpoint=self._ingest_endpoint,
-                               timeout=self._ingest_timeout)
+        try:
+            client = signalfx.SignalFx()
+            ingest = client.ingest(self._ingest_token, endpoint=self._ingest_endpoint,
+                                   timeout=self._ingest_timeout)
+        except Exception as e:
+            self._logger.error("An error occured when creating the ingest client: {0}".format(e))
+
         return ingest
```

Comment:
Looks like ingest would be undefined if an exception was thrown. You could reraise the exception with a bare `raise` after logging the error maybe. Or if you want to return `None` you could initialize `ingest = None` at the top outside of the try.